services/bundlejs/bundlejs-package.tester.js

```diff
@@ -0,0 +1,33 @@
+import { isFileSize } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('bundlejs/package (packageName)')
+  .timeout(20000)
```

Comment:
Before reviewing this in any detail, seeing these timeouts made me want to look into API peformance.

For shields we really need the upstream API call to usually return in under 4 seconds for a badge to be viable. GitHub (which is the place where badges are most commonly viewed) has a hard limit of 4 seconds on an image request so if it takes us >4 seconds to render a badge, they just don't get displayed.

Having tried making a few API calls, it seems sometimes the request does take a really long time (23 seconds was the longest I observed), but then the second call to the same URL is quick. I guess the API is slow if the request is a cache miss but then fast once the data is cached.

@okikio - are you also to provide any info on this? Also there any rate limits on using this API, etc?

Reply:
That's due to the bundlejs API bundling the package 📦. 

By default bundlejs caches the bundle on request but we set the TTL to 1 day to allow for new versions to propagate quickly.

Reply:
This feels like a really similar issue to what we had with socket.dev in https://github.com/badges/shields/pull/7727

For a really popular package its probably not a huge issue if one request in 24 hours is really slow. For smaller less popular packages, its probably more of an issue. I don't think I have a really clear answer right now, but it might be a good one for us to discuss next time we catch up @calebcartwright

Reply:
For smaller packages it's usually rather fast, it the larger packages that are the largest slow down

Reply:
Theoretically bundlejs should be faster than bundlephobia who is using webpack. @okikio Have you looked into the performance bottleneck of the service?

Reply:
Bundlephobia is bundling all the packages and storing them permanently, I'm currently not doing that because bundlejs allows for custom code so you can get treeshaken bundle sizes, and I just hadn't gotten around to setting up a permanent database for storing all the bundle size data for simple straight forward bundles, e.g. react@18.0.0

I'll get on it as soon as I can

Reply:
Bundlephobia also caches tree-shaking size of each export. They are limited and inaccurate most of the time due to the webpack config.

Reply:
Yeah, in the future I hope to also do treeshake bundle analysis for all the exports in a package, right now you have to manually give it a the functions and methods you'd like tteeshaken

Reply:
Hello. After some discussion in https://github.com/badges/shields/issues/9174 we've settled on a pattern for handling services that work in this way and implemented it in https://github.com/badges/shields/pull/9233

I appreciate this has take a while, but would you be able to rebase this PR on master and update it use the approach shown in code example in https://github.com/badges/shields/pull/9233 ?

Reply:
Sure.

Reply:
Yeah, don't use `no-cache` if you can avoid it

Reply:
> Yeah, don't use `no-cache` if you can avoid it

Any ETA for the fix? Last time I checked the package size was still incorrect. Maybe cache with `q` plus other search params?

Reply:
I forgot to leave a message here this has now been fixed

Reply:
> bundlejs now permanently caches the bundle size for packages

Does this mean that the timeout behavior will only ever occur for first-time packages, and that e.g. the size for jquery is permanently stored? If so then we could remove these timeouts from the code (it'll never get that far anyway given the timeout short circuit in the request fetching)

Reply:
Yeah I think given this badge will always return a response of some description (even if that is `bundling`) within 3 seconds, we should be able to remove all these `.timeout(20000)`s .